Sources/Plasma/FeatureLib/pfPatcher/pfPatcher.cpp

```diff
@@ -476,55 +480,78 @@ void pfPatcherWorker::Run()
 
 void pfPatcherWorker::ProcessFile()
 {
+    plFileName clientDir = plFileSystem::GetCurrentAppPath().StripFileName();
+
     do {
         NetCliFileManifestEntry& entry = fQueuedFiles.front();
+        fQueuedFiles.pop_front();
 
         // eap sucks
-        plFileName clName = ST::string::from_wchar(entry.clientName);
+        plFileName clName = plFileName(ST::string::from_wchar(entry.clientName)).Normalize();
         ST::string dlName = ST::string::from_wchar(entry.downloadName);
 
+        // Only operate on files if they are contained within the client directory...
+        plFileName clAbsPath = clName.AbsolutePath();
+        if (!clAbsPath.AsString().starts_with(clientDir.AsString())) {
```

Comment:
I considered that solution as well but eventually decided I preferred outright deletion. I know it's sketchy to allow deletion, but we're already downloading arbtirary Python code in python.pak, which is significantly more dangerous IMO. This changeset is not essential for Gehn 25, so we can defer until later if needed